integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/GCPUtils.java

```diff
@@ -0,0 +1,251 @@
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.apache.spark.SparkContext;
+import org.apache.spark.sql.execution.SparkPlan;
+import org.apache.spark.sql.execution.WholeStageCodegenExec;
+
+/** Util to extract values from GCP environment */
+public class GCPUtils {
+
+  private static final String PROJECT_ID_URI =
+      "http://metadata.google.internal/computeMetadata/v1/project/project-id";
+  private static final String BATCH_ID_URI =
+      "http://metadata.google.internal/computeMetadata/v1/instance/attributes/dataproc-batch-id";
+  private static final String BATCH_UUID_URI =
+      "http://metadata.google.internal/computeMetadata/v1/instance/attributes/dataproc-batch-uuid";
+  private static final String SESSION_ID_URI =
+      "http://metadata.google.internal/computeMetadata/v1/instance/attributes/dataproc-session-id";
+  private static final String SESSION_UUID_URI =
+      "http://metadata.google.internal/computeMetadata/v1/instance/attributes/dataproc-session-uuid";
+  private static final String CLUSTER_UUID_URI =
+      "http://metadata.google.internal/computeMetadata/v1/instance/attributes/dataproc-cluster-uuid";
+  private static final String DATAPROC_REGION_URI =
+      "http://metadata.google.internal/computeMetadata/v1/instance/attributes/dataproc-region";
+  private static final String DATAPROC_CLASSPATH = "/usr/local/share/google/dataproc/lib";
+  private static final CloseableHttpClient HTTP_CLIENT;
+  private static final String SPARK_YARN_TAGS = "spark.yarn.tags";
+  private static final String SPARK_DRIVER_HOST = "spark.driver.host";
+  private static final String SPARK_APP_ID = "spark.app.id";
+  private static final String SPARK_APP_NAME = "spark.app.name";
+  private static final String SPARK_MASTER = "spark.master";
+  private static final String JOB_ID_PREFIX = "dataproc_job_";
+  private static final String JOB_UUID_PREFIX = "dataproc_uuid_";
+  private static final String METADATA_FLAVOUR = "Metadata-Flavor";
+  private static final String GOOGLE = "Google";
+  private static final String SPARK_DIST_CLASSPATH = "SPARK_DIST_CLASSPATH";
+
+  enum ResourceType {
+    CLUSTER,
+    BATCH,
+    INTERACTIVE,
+    UNKNOWN
+  }
+
+  static {
+    RequestConfig config =
+        RequestConfig.custom()
+            .setConnectTimeout(100)
+            .setConnectionRequestTimeout(100)
+            .setSocketTimeout(100)
+            .build();
+    HTTP_CLIENT = HttpClients.custom().setDefaultRequestConfig(config).build();
+  }
+
+  public static boolean isGCPRuntime() {
+    String sparkDistClasspath = Environment.getEnvironmentVariable(SPARK_DIST_CLASSPATH);
+    return (sparkDistClasspath != null && sparkDistClasspath.contains(DATAPROC_CLASSPATH));
+  }
+
+  public static void setDataprocSpecificFacets(
```

Comment:
wouldn't be better for this method to return a map instead filling the map provided as argument? If not, it may be worth adding feature flag in config enabling this to be turned ON/OFF.

Reply:
Right, will change this to return a map instead.

Reply:
Done